Upgrade to Chroma Vector database API v2

dev-jonghoonpark opened the pull request:
ChromaDB 1.0.0 has recently been released.
- [GitHub Release](https://github.com/chroma-core/chroma/releases/tag/1.0.0)
- [Docker Hub](https://hub.docker.com/r/chromadb/chroma/tags?name=1.0.0)

With this release, the v1 APIs have been completely deprecated.
To support the new v2 APIs, we need to upgrade API implementation.

related issues
- https://github.com/spring-projects/spring-ai/issues/2648
- https://github.com/spring-projects/spring-ai/issues/2252

markpollack commented:
Thank you very much!

dev-jonghoonpark commented:
@markpollack I have completed the rebase to resolve the conflict.

dev-jonghoonpark commented:
recently, [testcontainers-java 1.21.0](https://github.com/testcontainers/testcontainers-java/releases/tag/1.21.0) is released 
this version include support of new version chroma. (related pr: https://github.com/testcontainers/testcontainers-java/pull/10170)

By upgrading the version of testcontainers-java, we can remove the temporal overridden wait strategy in this pr:
```java
static {
    chromaContainer.waitingFor(new AbstractWaitStrategy() {
        @Override
        protected void waitUntilReady() {
            Wait.forHttp("/api/v2/heartbeat");
        }
    });
}
```

If you want, I can update the PR. But it’s fine to leave it as is.

@ilayaperumalg

ilayaperumalg commented:
@dev-jonghoonpark Please go ahead and update the PR with your suggestion. 
@ElishebaW We target to merge this soon and be available for RC1. Thanks for your patience on this.

ElishebaW commented:
@ilayaperumalg  Thank you so much!

dev-jonghoonpark commented:
- removed the temporal overridden wait strategy
- removed `@Disabled` in `ChromaVectorStoreAutoConfigurationIT.addAndSearchWithFilters` (related pr: https://github.com/spring-projects/spring-ai/pull/2391)
  - The changes included in this PR seem to fix the issues that were previously present, and everything appears to be working correctly now.

ilayaperumalg commented:
@dev-jonghoonpark Thank you once again for the PR upgrading Chroma vector database to use v2 API. Rebased and merged as 09a6a6e1e
